tf_agents/environments/rnd_wrapper.py

```diff
@@ -0,0 +1,102 @@
+# coding=utf-8
+# Copyright 2018 The TF-Agents Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Wrapper implementing Random Network Distillation (RND)."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import gin
+import gym
+import gym.spaces
+import numpy as np
+import tensorflow as tf
+
+from tf_agents import specs
+from tf_agents.environments import wrappers
+from tf_agents.networks import rnd_network
+from tf_agents.trajectories import time_step as ts
+from tf_agents.trajectories import trajectory
+from tensorflow.python.util import nest  # pylint:disable=g-direct-tensorflow-import  # TF internal
+
+
+@gin.configurable
+class RNDWrapper(wrappers.PyEnvironmentBaseWrapper):
```

Comment:
I think the main reason for your issue is that the networks are in this wrapper, which then gets put into a pyfunc which then tries to evaluate the networks.

Two options I see, either move the rnd networks to the agent, and do the calculations there, or create a new tfenv wrapper which can do the extra RND work.

Note that the agent needs to be able to update the variables for the predictor so it may make more sense to have that in the agent.

What are your thoughts?